crates/ere-sp1/src/compile.rs

```diff
@@ -70,7 +72,94 @@ pub fn compile(guest_directory: &Path) -> Result<Vec<u8>, CompileError> {
         .map_err(|e| CompileError::CargoProveBuild {
             cwd: guest_directory.to_path_buf(),
             source: e,
-        })?;
+        });
+    match result {
+        Ok(status) => Ok((status, output_directory.join("guest.elf"))),
+        Err(err) => Err(err),
+    }
+}
+
+fn stock_rust_compile(
+    guest_directory: &Path,
+    output_directory: &Path,
+    program_name: &String,
+    toolchain: &String,
+) -> Result<(ExitStatus, PathBuf), CompileError> {
+    info!(
+        "Running `cargo build` (toolchain `{}`) → dir: {}",
+        toolchain,
+        output_directory.display(),
+    );
+
+    let rust_flags = [
+        "-C",
+        "passes=lower-atomic", // Only for rustc > 1.81
+        "-C",
+        "link-arg=-Ttext=0x00201000",
+        "-C",
+        "link-arg=--image-base=0x00200800",
+        "-C",
+        "panic=abort",
+        "--cfg",
+        "getrandom_backend=\"custom\"",
```

Comment:
does this need to be a cfg flag? is it possible to add the getrandom dep and then set feature flag?

Reply:
I think its possible to also have a get_random dep and set feature flag to "custom" -- though I'm not super convinced its strictly better than passing it in as a comand line arg (the former is more explicit)